### PR TITLE
RT11xx HWInfo driver

### PR DESCRIPTION
**Pull request by danieldegrasse**
RT11xx uses the same system reset controller IP as RT10xx, with changes made to accommodate the M4 core being able to reset independently of the M7 core. This PR updates the `hwinfo_mcux_src` driver to work with RT11xx system reset controller, and enables the hwinfo driver on the RT11xx SOC. The NXP hal revision must be updated to pull in the soc specific system reset controller driver that the RT11xx SOC uses.

Tested using `tests/drivers/hwinfo/api`

### CODE REVIEW

**Review by alexanderwachter — COMMENTED** [review]
It seems that the current mcux driver does not have much in common with your addition.
Could it make sense to make a now source-file?

**Review by dleach02 — DISMISSED** [review]

**Review by mbolivar-nordic — COMMENTED** [review]
DT portions LGTM

**Review by mbolivar-nordic — COMMENTED** [review]
DT portions still LGTM

File: drivers/hwinfo/hwinfo_mcux_src_rev2.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+#include <fsl_soc_src.h>
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1, "No nxp,imx-src compatible device found");
+
+int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
+{
+	uint32_t flags = 0;
+	uint32_t reason = SRC_GetResetStatusFlags((SRC_Type *)DT_INST_REG_ADDR(0));
+
+	if (reason & (kSRC_M7CoreIppUserResetFlag | kSRC_M4CoreIppUserResetFlag)) {
```
Comment (alexanderwachter) [inline drivers/hwinfo/hwinfo_mcux_src_rev2.c:22]: Are the reset cause for both cores always the same?
If not, how does the user know which cause applies to which core?

Comment (danieldegrasse) [reply drivers/hwinfo/hwinfo_mcux_src_rev2.c:22]: The reset status flags for each core should be the same-this is a good point. I will change the driver to use one flag or the other based on which core is being targeted during compilation.

**Review by dleach02 — APPROVED** [review]

**Review by alexanderwachter — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**